ledger-core/virtual/execute/execute_test.go

```diff
@@ -562,4 +562,53 @@ func TestSendVStateReportWithMissingState_IfConstructorWasInterruptedBeforeRunne
 		close(vStateReportRecv)
 		t.Error("timeout")
 	}
+
+	mc.Finish()
+}
+
+func TestSMExecute_StopWithoutMessagesIfPulseChangedBeforeOutgoing(t *testing.T) {
+	t.Log("C5101")
```

Comment:
I think you should make an integration test, check pendings, delegation token, vstatereport

Reply:
I can't write full integration test because server does not allow to run migration in right step. But checks for VStateReport was added to test.